webapp/tests/karma/unit/services/contact-view-model-generator.js

```diff
@@ -107,14 +109,24 @@ describe('ContactViewModelGenerator service', () => {
     KarmaUtils.restore(GetDataRecords);
   });
 
+   // REVIEWER: I suspect this should be implemented using Q, but I can't work out how
+  function waitForModelToLoad(model) {
```

Comment:
I've hit this problem twice now testing with karma. Unless there is actually a good better solution to this, maybe generalize it and put it somewhere shared? I'd use it.

`function waitForQPromise(promise)` ?

Reply:
Have you done similar elsewhere?  If you have other examples, we can pull code out and share it.

Reply:
I was able to fix it each time by refactoring - either injecting `Promise` in place ot `Q` or using the `Q.deferred` interface. I never found a solution like yours.